bin/update_hostname_validator.php

```diff
@@ -177,9 +177,12 @@ function getNewValidTlds($string)
 function getPunycodeDecoder()
 {
     if (function_exists('idn_to_utf8')) {
-        return function ($domain) {
-            return idn_to_utf8($domain, 0, INTL_IDNA_VARIANT_UTS46);
-        };
+        if (defined('INTL_IDNA_VARIANT_UTS46')) {
+            return function ($domain) {
+                return idn_to_utf8($domain, 0, INTL_IDNA_VARIANT_UTS46);
+            };
+        }
+        return 'idn_to_utf8';
```

Comment:
We don't need to do the check here, as this script is only executed by maintainers and by Travis (which has an up-to-date libicu already).